detectable/src/main/java/com/synopsys/integration/detectable/detectables/npm/packagejson/CombinedPackageJsonExtractor.java

```diff
@@ -87,6 +86,27 @@ public CombinedPackageJson constructCombinedPackageJson(String rootJsonPath, Str
         return combinedPackageJson;
     }
 
+
+    /**
+     * Takes an absolute path to a workspace and converts it to a relative one for
+     * future comparisons with contents of the package-lock.json file.
+     * 
+     * @param combinedPackageJson the combined package json we are constructing
+     * @param projectRoot         Path to the root of the project
+     * @param convertedWorkspace  An absolute path to a workspace with wildcards
+     *                            replaced
+     */
+    private void addRelativeWorkspace(CombinedPackageJson combinedPackageJson, String projectRoot,
```

Comment:
Just curious - was the motivation here to avoid using File/Path related operations like relativize, getAbsolutePath to obtain the relative path and stick to string operations? I guess these paths are all strings everywhere else so it makes sense if you wanted to avoid that.

Reply:
No real strong motivation to go with what I went with. I suppose I just understand the code a bit better if I control it from start to finish. We could discuss an alternate implementation if you think something exists that could streamline this. Basically before this call the code has an absolute path to the workspace. This was pretty much how the information came into this class and it does make reading the various files easier. After this call we switch over to relative workspaces as we are changing from opening files to doing comparisons, so I can know we are dealing with packages/a or whatever and can do comparisons against similar lines in the package-lock.json where I can't do an equals on /Users/dterry/...../packages/a for example.

Reply:
Ah got it. I think the current implementation is reasonable and consistent with the rest of the pattern followed in this flow. I was just curious to know if avoiding the overhead of file operations was the motivation